source/Cosmos.System2/Graphics/SVGAIICanvas.cs

```diff
@@ -1,398 +1,169 @@
+        /// </summary>
+        private Mode mode;
+        public override Mode Mode
+        {
+            get { return mode; }
+            set
+            {
+                mode = value;
+                if (SendDebug) { debugger.SendInternal($"Set mode property to {value}"); }
+                SetMode(value);
+            }
+        }
+
+        /// <summary>
+        /// Create new SVGA canvas with default mode
+        /// </summary>
+        public SVGAIICanvas() { driver = new VMWareSVGAII(); this.Mode = DefaultGraphicMode; }
+
+        /// <summary>
+        /// Create new SVGA canvas with specified mode
+        /// </summary>
+        /// <param name="mode"></param>
+        public SVGAIICanvas(Mode mode) { driver = new VMWareSVGAII(); this.Mode = mode; }
+
+        // set graphics mode
+        private void SetMode(Mode mode)
+        {
+            this.mode = mode;
+            if (!CheckIfModeIsValid(mode)) { throw new Exception($"Unsupported mode {mode}"); }
+            if (driver == null) { driver = new VMWareSVGAII(); }
+            driver.SetMode((uint)mode.Width, (uint)mode.Height, (uint)mode.Depth);
+
+        }
+
+        /// <summary>
+        /// Clear the screen with specified color
+        /// </summary>
+        /// <param name="color"></param>
+        public override void Clear(Color color) { driver.Clear((uint)color.ToArgb()); }
+
+        /// <summary>
+        /// Disable video driver
+        /// </summary>
+        public override void Disable() { driver.Disable(); }
+
+        /// <summary>
+        /// Render canvas onto screen
+        /// </summary>
+        public override void Display() { driver.Update(0, 0, (uint)Mode.Width, (uint)Mode.Height); }
+
+        /// <summary>
+        /// Draw filled rectangle with color
+        /// </summary>
+        /// <param name="color"></param>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        public override void DrawFilledRectangle(Color color, int x, int y, int width, int height) { driver.Fill((uint)x, (uint)y, (uint)width, (uint)height, (uint)color.ToArgb()); }
+
+        public override void DrawImage(Image image, int x, int y)
+        {
+            for (int i = 0; i < image.Width * image.Height; i++)
+            {
+                DrawPoint(image.RawData[i], x + (i % image.Width), y + (i / image.Width));
+            }
+        }
+
+        /// <summary>
+        /// Draw point with color at specified position
+        /// </summary>
+        /// <param name="color"></param>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        public override void DrawPoint(Color color, int x, int y)
+        {
+            if (x < 0 || x >= Mode.Width || y < 0 || y >= Mode.Height) { if (SendDebug) { debugger.SendInternal("Tried to draw pixel out of bounds at " + x.ToString() + ", " + y.ToString()); } return; }

```

Comment:
Is there a reason why DrawPoint does not just throw an error when trying to draw at an invalid location?

Reply:
The same goes for the remaining functions in this file.